tests/test_env_example.py

```
@@ -0,0 +1,345 @@
+    ipv4 = re.match(r"^\d{1,3}(\.\d{1,3}){3}$", host)
+    if ipv4:
+        return all(0 <= int(part) <= 255 for part in host.split("."))
+    # Hostname pattern per RFC 1123 (simplified)
+    return bool(re.match(r"^(?=.{1,253}$)([a-zA-Z0-9-]{1,63}\.)*[a-zA-Z0-9-]{1,63}$", host))
+
+def parse_endpoint_host_port(endpoint: str) -> tuple[str, int]:
+    """
+    Expect format 'host:port'
+    """
+    if ":" not in endpoint:
+        raise ValueError("Endpoint must contain host:port")  # noqa
```

Ruff (0.12.2)

Unused blanket `noqa` directive

Remove unused `noqa` directive

(RUF100)